### PR TITLE
Core: Support IncrementalChangelogScan with deletes.

### PR DESCRIPTION
**Pull request by Reo-LEI**
This PR is trying to support `IncrementalChangelogScan` with deletes after https://github.com/apache/iceberg/pull/5382

Any feedback is welcome！
cc @rdblue @aokolnychyi @flyrain @stevenzwu @kbendick @chenjunjiedada

### CODE REVIEW

**Comment by aokolnychyi** [comment]
I had a local PoC for this which was blocked by some changes around sequence numbers. I'll try to find some time to review this PR later this week.

**Comment by nastra** [comment]
@Reo-LEI @aokolnychyi any updates on this one? Would be great to get this in

**Comment by aokolnychyi** [comment]
Sorry for the delay on this one. I will get to it as soon as I can.

File: core/src/main/java/org/apache/iceberg/BaseIncrementalChangelogScan.java [hunk 1]
```diff
@@ -105,79 +123,161 @@ private Deque<Snapshot> orderedChangelogSnapshots(Long fromIdExcl, long toIdIncl
...
+        return new BaseDeletedDataFileScanTask(
+            snapshotOrdinals.get(commitSnapshotId),
+            commitSnapshotId,
+            dataFile,
+            deleteFiles,
+            context.schemaAsString(),
+            context.specAsString(),
+            context.residuals());
+      } else {
+        // ignore removed data files.
```
Comment (alexjo2144) [inline core/src/main/java/org/apache/iceberg/BaseIncrementalChangelogScan.java:215]: This case where the commit snapshot id doesn't exist in `snapshotOrdinals` didn't seem to be possible before, what changed?

**Review by alexjo2144 — COMMENTED** [review]
Just a couple questions for my understanding:

**Comment by manuzhang** [comment]
@Reo-LEI and @aokolnychyi, are you still working on this?

**Comment by manuzhang** [comment]
I just submitted https://github.com/apache/iceberg/pull/9888

**Pull request closed without merging**